Lib/test/test_free_threading/test_set.py

```diff
@@ -36,6 +36,70 @@ def repr_set():
             for set_repr in set_reprs:
                 self.assertIn(set_repr, ("set()", "{1, 2, 3, 4, 5, 6, 7, 8}"))
 
+    def test_contains_mutate(self):
+        """Test set contains operation combined with mutation."""
+        barrier = Barrier(2)
+        s = set()
+        done = False
+
+        NUM_ITEMS = 2_000
+        NUM_LOOPS = 20
+
+        def read_set():
+            barrier.wait()
+            while not done:
+                for i in range(NUM_ITEMS):
+                    item = i >> 1
+                    result = item in s
+
+        def mutate_set():
+            nonlocal done
+            barrier.wait()
+            for i in range(NUM_LOOPS):
+                s.clear()
+                for j in range(NUM_ITEMS):
+                    s.add(j)
+                for j in range(NUM_ITEMS):
+                    s.discard(j)
+                # executes the set_swap_bodies() function
+                s.__iand__(set(k for k in range(10, 20)))
+            done = True
+
+        threads = [Thread(target=read_set), Thread(target=mutate_set)]
+        for t in threads:
+            t.start()
+        for t in threads:
+            t.join()
+
+    def test_contains_frozenset(self):
+        barrier = Barrier(3)
+        done = False
+
+        NUM_ITEMS = 2_000
+        NUM_LOOPS = 20
+
+        s = frozenset()
+        def make_set():
```

Comment:
Why is the `make_set` here in a thread? It just assigns the same frozenset to `s` each time.

Maybe change `make_set` so that it creates a different `frozenset` each time?

Reply:
I don't understand this comment.  The `make_set()` binds `s` to a new frozenset object (at least I intend it to).

Reply:
The `s` is indeed updated with a new `frozenset`, but it is a `frozenset` with the same elements each time. I was thinking about something like
```
def make_set():
            nonlocal s
            barrier.wait()
            num_elements = NUM_ITEMS
            while not done:
               num_elements = (num_elements  + 17 %) NUM_ITEMS
                s = frozenset(range(num_elements))
```
But what are we testing here. There is no concurrent mutation of the `s`, so this test is not specific to the free-threading build I think?